EventFilter/L1TRawToDigi/plugins/implementations_stage2/ZDCUnpacker.cc

```diff
@@ -0,0 +1,71 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "EventFilter/L1TRawToDigi/plugins/UnpackerFactory.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/CaloTools.h"
+
+#include "L1TObjectCollections.h"
+
+// #include "L1TStage2Layer2Constants.h"
+#include "ZDCUnpacker.h"
+#include "GTSetup.h"
+
+namespace l1t {
+  namespace stage2 {
+    ZDCUnpacker::ZDCUnpacker() : EtSumZDCCopy_(0) {}
+
+    bool ZDCUnpacker::unpack(const Block& block, UnpackerCollections* coll) {
+      using namespace l1t::stage2;
+      uint32_t zdc_mask = 0x3FF;
+      LogDebug("L1T") << "Block ID  = " << block.header().getID() << " size = " << block.header().getSize();
+
+      int nBX =
+          int(ceil(block.header().getSize() / zdc::nOutputFramePerBX));  // Since there 6 frames per demux output event

```

Comment:
This looks like a division between two integers, `lock.header().getSize()` and `zdc::nOutputFramePerBX`.
If this is indeed the case, the result is automatically truncated towards zero, and the use of the `ceil()` function will not do anything.
If the intent is to round the result of the division _up_, it should be written as
```suggestion
          int(std::ceil(static_cast<float>(block.header().getSize()) / zdc::nOutputFramePerBX));
          // Since there 6 frames per demux output event
```
or
```suggestion
          (block.header().getSize() + zdc::nOutputFramePerBX - 1) / zdc::nOutputFramePerBX;
          // Since there 6 frames per demux output event

```

Reply:
Not that this excuses it, but I believe this is how it is done in the current EtSum unpacker: https://github.com/cms-sw/cmssw/blob/99f2750c476c243522d4599697496b1af4c29ce7/EventFilter/L1TRawToDigi/plugins/implementations_stage2/EtSumUnpacker.cc#L20-L21